node/pkg/common/grpc.go

```diff
@@ -2,60 +2,137 @@ package common
 
 import (
 	"context"
+	"fmt"
 
 	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
 	grpc_zap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap"
 	grpc_ctxtags "github.com/grpc-ecosystem/go-grpc-middleware/tags"
 	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
 	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
-// xForwardedForStreamServerInterceptor adds the `x-forwarded-for` metadata as a tag to cause it to be logged by grpc_zap.StreamServerInterceptor().
-// Note that `x-forwarded-for` can only be trusted if the latest hop in the proxy chain is trusted.
-// For JSON-Web requests, the latest hop is the guardian itself (`grpc-gateway`), which is listening on TCP and forwarding to the gRPC publicrpc UNIX socket.
-// This can be identified by `"peer.address": "@"` in the logs and `grpc-gateway` correctly sets the `x-forwarded-for` metadata.
-func xForwardedForStreamServerInterceptor(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
-	ctx := stream.Context()
+type GrpcLogDetail string
+
+const (
+	GrpcLogDetailNone    GrpcLogDetail = "false"
+	GrpcLogDetailMinimal GrpcLogDetail = "minimal"
+	GrpcLogDetailFull    GrpcLogDetail = "full"
+)
```

Comment:
Sorry, I keep coming back to this.  The semantics keep throwing me off.  But, maybe that's just me.  The following would be clearer to me:
```
type GrpcLogLevel string
const (
	GrpcLoggingNone    GrpcLogLevel = "none"
	GrpcLoggingMinimal GrpcLogLevel = "minimal"
	GrpcLoggingFull    GrpcLogLevel = "full"
)

Reply:
good suggestion